test/TestUtils.py

```diff
@@ -185,3 +186,36 @@ def test_expand_paths_vars(monkeypatch):
     monkeypatch.setenv('TEST_PATH', test_path)
     assert utils.expand_paths_vars(['~']) == [os.path.expanduser('~')]
     assert utils.expand_paths_vars(['$TEST_PATH']) == [test_path]
+
+
+def test_get_yaml_files(monkeypatch, capsys):
+    options_silent, _ = cli.get_config([])
+    options_verbose, _ = cli.get_config(['-v'])
+    sys_path = os.getenv('PATH')
+    sys_dir = os.getcwd()
+    test_dir = os.path.dirname(os.path.realpath(__file__))
+    role_path = os.path.join(test_dir, 'roles/test-role')
+
+    # Verbose and git command is absent
+    monkeypatch.setenv('PATH', '')
+    utils.get_yaml_files(options_verbose)
+    stdout = capsys.readouterr().out.split('\n')
+    assert stdout[0] == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert stdout[1].startswith('Warning: Failed to locate command:')
+    monkeypatch.setenv('PATH', sys_path)
+
+    # Verbose and directory is not in git repository
+    monkeypatch.chdir('/tmp')
+    utils.get_yaml_files(options_verbose)
+    stdout = capsys.readouterr().out.split('\n')
+    assert stdout[0] == 'Discovering files to lint: git ls-files *.yaml *.yml'
+    assert stdout[1].startswith('Warning: Failed to discover yaml files to lint using git:')
+    monkeypatch.chdir(sys_dir)
+
+    # Silent and returned dict is not empty
+    monkeypatch.chdir(role_path)
+    files = utils.get_yaml_files(options_silent)
+    stdout = capsys.readouterr().out
+    assert not stdout
+    assert len(files)
+    monkeypatch.chdir(sys_dir)
```

Comment:
You don't need to restore anything. `monkeypatch` will revert the change upon exiting the test automatically.